utils/python/CIME/create_test.py

```diff
@@ -112,22 +119,19 @@ def __init__(self, test_names,
                        "Missing baseline comparison directory %s" % full_baseline_dir)
 
         # This is the only data that multiple threads will simultaneously access
-        # Each test has it's own index and setting/retrieving items from a list
-        # is atomic, so this should be fine to use without mutex
-        self._test_states = [ (INITIAL_PHASE, TEST_PASS_STATUS) ] * len(self._tests)
-        idx = 0
-        for test in self._tests:
-            test["state_idx"] = idx
-            idx += 1
+        # Each test has it's own value and setting/retrieving items from a dict
+        # is atomic, so this should be fine to use without mutex.
+        # Since the name-list phase can fail without aborting later phases, we
+        # need some extra state to remember tests that had namelist problems.
+        # name -> (phase, status, has_namelist_problem)
```

Comment:
I understand why you didn't like the old data structure. Maybe you'll like this one better.